src/commands/deploy/deploy.ts

```diff
@@ -62,6 +63,35 @@ async function deploy(actionContext: IActionContext, arg1: vscode.Uri | string |
         throw new Error(localize('pythonNotAvailableOnWindows', 'Python projects are not supported on Windows Function Apps. Deploy to a Linux Function App instead.'));
     }
 
+    const showCoreToolsWarningKey: string = 'showCoreToolsWarning';
+    if (getWorkspaceSetting<boolean>(showCoreToolsWarningKey)) {
```

Comment:
I would prefer that you move all of this code in a separate file. Maybe something like `showCoreToolsWarning.ts` that also exists in the deploy folder.

`deploy.ts` is already complicated code to look at so moving this out can help new viewers digest the code a lot easier.